### ISSUE TITLE
fails to detect correct ruby version

### ISSUE THREAD

**Issue by floh96** [Shopify/ruby-lsp#1517]
### Reproduction steps

When I try to start the ruby lsp server it fails with the following error message
`Failed to activate asdf environment: The Ruby LSP requires Ruby 3.0 or newer to run. This project is using 2.6.6. `

I'm using asdf as ruby version manager with the following setup:
I have set Ruby 2.6.6 as my global ruby version via _~/.tool-versions_.
The project is using ruby 3.2.2 which is set via _projectdir/.ruby-version_.
Asdf is configured to respect .ruby-version and asdf current inside the project directory says ruby 3.2.2 is the current ruby version.
Output of the command _/usr/bin/fish -i -c 'asdf exec ruby -rjson -e "STDERR.printf(%{RUBY_ENV_ACTIVATE%sRUBY_ENV_ACTIVATE}, JSON.dump({ env: ENV.to_h, ruby_version: RUBY_VERSION, yjit: defined?
(RubyVM::YJIT) }))"'_:

`RUBY_ENV_ACTIVATE{"env":{"PATH":"/home/fheberl/.asdf/installs/ruby/3.2.2/bin:/home/fheberl/Projekte/infrastructure:/home/fheberl/.local/bin:/home/fheberl/.asdf/shims:/home/fheberl/.asdf/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin","SHELL":"/usr/bin/fish","SESSION_MANAGER":"local/unix:@/tmp/.ICE-unix/4419,unix/unix:/tmp/.ICE-unix/4419","COLORTERM":"truecolor","XDG_MENU_PREFIX":"gnome-","TERM_PROGRAM_VERSION":"1.87.0","asdf_data_dir":"/home/fheberl/.asdf","omp_lastcommand":"asdf current","SSH_AUTH_SOCK":"/run/user/1000/keyring/ssh","MEMORY_PRESSURE_WRITE":"c29tZSAyMDAwMDAgMjAwMDAwMAA=","XMODIFIERS":"@im=ibus","DESKTOP_SESSION":"gnome","NO_AT_BRIDGE":"1","EDITOR":"code --wait --new-window","PWD":"/home/fheberl/Projekte/nix-wie-weg-de","LOGNAME":"fheberl","XDG_SESSION_DESKTOP":"gnome","XDG_SESSION_TYPE":"wayland","SYSTEMD_EXEC_PID":"4529","XAUTHORITY":"/run/user/1000/.mutter-Xwaylandauth.PWOBK2","VSCODE_GIT_ASKPASS_NODE":"/usr/share/code/code","ASDF_DEFAULT_TOOL_VERSIONS_FILENAME":".tool-versions","VSCODE_INJECTION":"1","GDM_LANG":"de_DE.UTF-8","HOME":"/home/fheberl","USERNAME":"fheberl","LANG":"de_DE.UTF-8","XDG_CURRENT_DESKTOP":"GNOME","POSH_SHELL_VERSION":"3.7.0","MEMORY_PRESSURE_WATCH":"/sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/session.slice/org.gnome.Shell@wayland.service/memory.pressure","VTE_VERSION":"7402","WAYLAND_DISPLAY":"wayland-0","GIT_ASKPASS":"/usr/share/code/resources/app/extensions/git/dist/askpass.sh","GNOME_TERMINAL_SCREEN":"/org/gnome/Terminal/screen/4d14c413_2032_45f6_9ee9_3321be0ed8be","ASDF_DATA_DIR":"/home/fheberl/.asdf","CHROME_DESKTOP":"code-url-handler.desktop","VSCODE_GIT_ASKPASS_EXTRA_ARGS":"","GNOME_SETUP_DISPLAY":":1","XDG_SESSION_CLASS":"user","TERM":"xterm-256color","ASDF_DIR":"/home/fheberl/.asdf","USER":"fheberl","VSCODE_GIT_IPC_HANDLE":"/run/user/1000/vscode-git-b2e5338e99.sock","GNOME_TERMINAL_SERVICE":":1.137","DISPLAY":":0","SHLVL":"4","QT_IM_MODULE":"ibus","POSH_THEME":"/home/fheberl/.oh-my-posh/themes/fish.json","XDG_RUNTIME_DIR":"/run/user/1000","VSCODE_GIT_ASKPASS_MAIN":"/usr/share/code/resources/app/extensions/git/dist/askpass-main.js","XDG_DATA_DIRS":"/home/fheberl/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share/:/usr/share/:/usr/share/code/resources/app/out/vs/workbench/contrib/terminal/browser/media/fish_xdg_data","omp_last_status_generation":"5","GDK_BACKEND":"x11","ASDF_CONFIG_FILE":"/home/fheberl/.asdfrc","GDMSESSION":"gnome","ORIGINAL_XDG_CURRENT_DESKTOP":"GNOME","DBUS_SESSION_BUS_ADDRESS":"unix:path=/run/user/1000/bus","RUBYLIB":"/home/fheberl/.asdf/plugins/ruby/rubygems-plugin","TERM_PROGRAM":"vscode"},"ruby_version":"3.2.2","yjit":"constant"}RUBY_ENV_ACTIVATE`

* OS: Fedora 39
* Shell: fish
* VSCode version: 1.87.0
* VSCode Ruby LSP version: 0.5.12
* Ruby version: 3.2.2

**Comment by floh96** [comment]
Hey @vinistock I started the debugger and in the ruby version variable the version is 2.6.6.
But if I run the command directly in the vscode terminal I get the result from my first comment (version = 3.2.2). 
So the difference must be how the extension executes the command?
Unfortunately I have no clue of typescript :/

**Comment by vinistock** [comment]
Was anything else printed to the output tab before it started erroring? It's hard to tell based on that trace alone.

**Comment by floh96** [comment]
@vinistock i reinstalled the extension and this is the complete output:

```
2024-05-07 10:25:26.405 [info] (nix-wie-weg.de) [Error - 10:25:26] Client Ruby LSP: connection to server is erroring.
write EPIPE
Shutting down server.
2024-05-07 10:25:26.405 [info] (nix-wie-weg.de) [Error - 10:25:26] Client Ruby LSP: connection to server is erroring.
write EPIPE
2024-05-07 10:25:26.405 [info] (nix-wie-weg.de) [Error - 10:25:26] Stopping server failed
2024-05-07 10:25:26.405 [info] (nix-wie-weg.de) Error: Client is not running and can't be stopped. It's current state is: starting
	at ws.shutdown (/home/fheberl/.vscode/extensions/shopify.ruby-lsp-0.5.21/out/extension.js:41:8468)
	at ws.stop (/home/fheberl/.vscode/extensions/shopify.ruby-lsp-0.5.21/out/extension.js:41:8047)
	at ws.stop (/home/fheberl/.vscode/extensions/shopify.ruby-lsp-0.5.21/out/extension.js:44:38355)
	at ws.handleConnectionError (/home/fheberl/.vscode/extensions/shopify.ruby-lsp-0.5.21/out/extension.js:43:63)
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2024-05-07 10:25:26.405 [info] (nix-wie-weg.de) [Error - 10:25:26] Server initialization failed.
2024-05-07 10:25:26.405 [info] (nix-wie-weg.de)   Message: write EPIPE
  Code: -32099 
2024-05-07 10:25:26.405 [info] (nix-wie-weg.de) [Error - 10:25:26] Ruby LSP client: couldn't create connection to server.
2024-05-07 10:25:26.405 [info] (nix-wie-weg.de)   Message: write EPIPE
  Code: -32099 
2024-05-07 10:25:26.405 [error] (nix-wie-weg.de) Error starting the server: write EPIPE
2024-05-07 10:25:26.406 [info] (nix-wie-weg.de) [Info  - 10:25:26] Connection to server got closed. Server will restart.
2024-05-07 10:25:26.406 [info] (nix-wie-weg.de) true
2024-05-07 10:25:26.409 [info] (nix-wie-weg.de) /bin/sh: Zeile 1: ruby-lsp: Kommando nicht gefunden.

2024-05-07 10:25:26.409 [info] (nix-wie-weg.de) [Error - 10:25:26] Server process exited with code 127.
2024-05-07 10:25:26.410 [info] (nix-wie-weg.de) [Error - 10:25:26] Client Ruby LSP: connection to server is erroring.
write EPIPE
Shutting down server.
2024-05-07 10:25:26.410 [info] (nix-wie-weg.de) [Error - 10:25:26] Client Ruby LSP: connection to server is erroring.
write EPIPE
2024-05-07 10:25:26.410 [info] (nix-wie-weg.de) [Error - 10:25:26] Stopping server failed
2024-05-07 10:25:26.410 [info] (nix-wie-weg.de) Error: Client is not running and can't be stopped. It's current state is: starting
	at ws.shutdown (/home/fheberl/.vscode/extensions/shopify.ruby-lsp-0.5.21/out/extension.js:41:8468)
	at ws.stop (/home/fheberl/.vscode/extensions/shopify.ruby-lsp-0.5.21/out/extension.js:41:8047)
	at ws.stop (/home/fheberl/.vscode/extensions/shopify.ruby-lsp-0.5.21/out/extension.js:44:38355)
	at ws.handleConnectionError (/home/fheberl/.vscode/extensions/shopify.ruby-lsp-0.5.21/out/extension.js:43:63)
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2024-05-07 10:25:26.410 [info] (nix-wie-weg.de) [Error - 10:25:26] Server initialization failed.
2024-05-07 10:25:26.410 [info] (nix-wie-weg.de)   Message: write EPIPE
  Code: -32099 
2024-05-07 10:25:26.410 [info] (nix-wie-weg.de) [Error - 10:25:26] Ruby LSP client: couldn't create connection to server.
2024-05-07 10:25:26.410 [info] (nix-wie-weg.de)   Message: write EPIPE
  Code: -32099 
2024-05-07 10:25:26.410 [info] (nix-wie-weg.de) [Error - 10:25:26] Restarting server failed
2024-05-07 10:25:26.410 [info] (nix-wie-weg.de)   Message: write EPIPE
  Code: -32099 
2024-05-07 10:25:26.410 [info] (nix-wie-weg.de) [Info  - 10:25:26] Connection to server got closed. Server will restart.
2024-05-07 10:25:26.410 [info] (nix-wie-weg.de) true
2024-05-07 10:25:26.413 [info] (nix-wie-weg.de) /bin/sh: Zeile 1: ruby-lsp: Kommando nicht gefunden.

2024-05-07 10:25:26.413 [info] (nix-wie-weg.de) [Error - 10:25:26] Server process exited with code 127.
2024-05-07 10:25:26.414 [info] (nix-wie-weg.de) [Error - 10:25:26] Client Ruby LSP: connection to server is erroring.
write EPIPE
Shutting down server.
2024-05-07 10:25:26.414 [info] (nix-wie-weg.de) [Error - 10:25:26] Client Ruby LSP: connection to server is erroring.
write EPIPE
2024-05-07 10:25:26.414 [info] (nix-wie-weg.de) [Error - 10:25:26] Stopping server failed
2024-05-07 10:25:26.414 [info] (nix-wie-weg.de) Error: Client is not running and can't be stopped. It's current state is: starting
	at ws.shutdown (/home/fheberl/.vscode/extensions/shopify.ruby-lsp-0.5.21/out/extension.js:41:8468)
	at ws.stop (/home/fheberl/.vscode/extensions/shopify.ruby-lsp-0.5.21/out/extension.js:41:8047)
	at ws.stop (/home/fheberl/.vscode/extensions/shopify.ruby-lsp-0.5.21/out/extension.js:44:38355)
	at ws.handleConnectionError (/home/fheberl/.vscode/extensions/shopify.ruby-lsp-0.5.21/out/extension.js:43:63)
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2024-05-07 10:25:26.414 [info] (nix-wie-weg.de) [Error - 10:25:26] Server initialization failed.
2024-05-07 10:25:26.414 [info] (nix-wie-weg.de)   Message: write EPIPE
  Code: -32099 
2024-05-07 10:25:26.414 [info] (nix-wie-weg.de) [Error - 10:25:26] Ruby LSP client: couldn't create connection to server.
2024-05-07 10:25:26.414 [info] (nix-wie-weg.de)   Message: write EPIPE
  Code: -32099 
2024-05-07 10:25:26.414 [info] (nix-wie-weg.de) [Error - 10:25:26] Restarting server failed
2024-05-07 10:25:26.414 [info] (nix-wie-weg.de)   Message: write EPIPE
  Code: -32099 
2024-05-07 10:25:26.415 [info] (nix-wie-weg.de) [Info  - 10:25:26] Connection to server got closed. Server will restart.
2024-05-07 10:25:26.415 [info] (nix-wie-weg.de) true
2024-05-07 10:25:26.418 [info] (nix-wie-weg.de) /bin/sh: Zeile 1: ruby-lsp: Kommando nicht gefunden.

2024-05-07 10:25:26.418 [info] (nix-wie-weg.de) [Error - 10:25:26] Server process exited with code 127.
2024-05-07 10:25:26.418 [info] (nix-wie-weg.de) [Error - 10:25:26] Client Ruby LSP: connection to server is erroring.
write EPIPE
Shutting down server.
2024-05-07 10:25:26.418 [info] (nix-wie-weg.de) [Error - 10:25:26] Client Ruby LSP: connection to server is erroring.
write EPIPE
2024-05-07 10:25:26.418 [info] (nix-wie-weg.de) [Error - 10:25:26] Stopping server failed
2024-05-07 10:25:26.418 [info] (nix-wie-weg.de) Error: Client is not running and can't be stopped. It's current state is: starting
	at ws.shutdown (/home/fheberl/.vscode/extensions/shopify.ruby-lsp-0.5.21/out/extension.js:41:8468)
	at ws.stop (/home/fheberl/.vscode/extensions/shopify.ruby-lsp-0.5.21/out/extension.js:41:8047)
	at ws.stop (/home/fheberl/.vscode/extensions/shopify.ruby-lsp-0.5.21/out/extension.js:44:38355)
	at ws.handleConnectionError (/home/fheberl/.vscode/extensions/shopify.ruby-lsp-0.5.21/out/extension.js:43:63)
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2024-05-07 10:25:26.418 [info] (nix-wie-weg.de) [Error - 10:25:26] Server initialization failed.
2024-05-07 10:25:26.418 [info] (nix-wie-weg.de)   Message: write EPIPE
  Code: -32099 
2024-05-07 10:25:26.418 [info] (nix-wie-weg.de) [Error - 10:25:26] Ruby LSP client: couldn't create connection to server.
2024-05-07 10:25:26.418 [info] (nix-wie-weg.de)   Message: write EPIPE
  Code: -32099 
2024-05-07 10:25:26.419 [info] (nix-wie-weg.de) [Error - 10:25:26] Restarting server failed
2024-05-07 10:25:26.419 [info] (nix-wie-weg.de)   Message: write EPIPE
  Code: -32099 
2024-05-07 10:25:26.419 [info] (nix-wie-weg.de) [Info  - 10:25:26] Connection to server got closed. Server will restart.
2024-05-07 10:25:26.419 [info] (nix-wie-weg.de) true
2024-05-07 10:25:26.422 [info] (nix-wie-weg.de) /bin/sh: Zeile 1: ruby-lsp: Kommando nicht gefunden.

2024-05-07 10:25:26.422 [info] (nix-wie-weg.de) [Error - 10:25:26] Server process exited with code 127.
2024-05-07 10:25:26.423 [info] (nix-wie-weg.de) [Error - 10:25:26] Client Ruby LSP: connection to server is erroring.
write EPIPE
Shutting down server.
2024-05-07 10:25:26.423 [info] (nix-wie-weg.de) [Error - 10:25:26] Client Ruby LSP: connection to server is erroring.
write EPIPE
2024-05-07 10:25:26.423 [info] (nix-wie-weg.de) [Error - 10:25:26] Stopping server failed
2024-05-07 10:25:26.423 [info] (nix-wie-weg.de) Error: Client is not running and can't be stopped. It's current state is: starting
	at ws.shutdown (/home/fheberl/.vscode/extensions/shopify.ruby-lsp-0.5.21/out/extension.js:41:8468)
	at ws.stop (/home/fheberl/.vscode/extensions/shopify.ruby-lsp-0.5.21/out/extension.js:41:8047)
	at ws.stop (/home/fheberl/.vscode/extensions/shopify.ruby-lsp-0.5.21/out/extension.js:44:38355)
	at ws.handleConnectionError (/home/fheberl/.vscode/extensions/shopify.ruby-lsp-0.5.21/out/extension.js:43:63)
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2024-05-07 10:25:26.424 [info] (nix-wie-weg.de) [Error - 10:25:26] Server initialization failed.
2024-05-07 10:25:26.424 [info] (nix-wie-weg.de)   Message: write EPIPE
  Code: -32099 
2024-05-07 10:25:26.424 [info] (nix-wie-weg.de) [Error - 10:25:26] Ruby LSP client: couldn't create connection to server.
2024-05-07 10:25:26.424 [info] (nix-wie-weg.de)   Message: write EPIPE
  Code: -32099 
2024-05-07 10:25:26.424 [info] (nix-wie-weg.de) [Error - 10:25:26] Restarting server failed
2024-05-07 10:25:26.424 [info] (nix-wie-weg.de)   Message: write EPIPE
  Code: -32099 
2024-05-07 10:25:26.424 [info] (nix-wie-weg.de) [Error - 10:25:26] The Ruby LSP server crashed 5 times in the last 3 minutes. The server will not be restarted. See the output for more information.
2024-05-07 10:25:26.425 [info] (nix-wie-weg.de) /bin/sh: Zeile 1: ruby-lsp: Kommando nicht gefunden.

2024-05-07 10:25:26.425 [info] (nix-wie-weg.de) [Error - 10:25:26] Server process exited with code 127.
```

**Comment by vinistock** [comment]
We released [v0.7.0](https://github.com/Shopify/ruby-lsp/releases/tag/vscode-ruby-lsp-v0.7.0) with a few ASDF related fixes. Can you please try it and let us know if you find any issues?

**Comment by floh96** [comment]
@vinistock version 0.7.1 works :tada: thanks for your help
